app/src/main/java/net/programmierecke/radiodroid2/alarm/AlarmReceiver.java

```diff
@@ -199,4 +203,72 @@ protected void onPostExecute(String result) {
             }
         }.execute();
     }
+
+    int currentVolume;
+    int minVolume;
```

Comment:
Could be move into the method.

Reply:
Doesn't look like these can be as I get a build error:

```
error: local variables referenced from an inner class must be final or effectively final
```

Reply:
You can move `currentVolume` to the runnable itself.

Reply:
👍 fixed in d348221

Reply:
you can move `minVolume` too, but better just tackle the core problem I described above - you'd shuffle the code more then once =)